clippy_lints/src/methods/unnecessary_lazy_eval.rs

```diff
@@ -32,6 +32,16 @@ pub(super) fn lint<'tcx>(
                 } else {
                     "unnecessary closure used to substitute value for `Result::Err`"
                 };
+                let applicability = if body
+                    .params
+                    .iter()
+                    .all(|param| matches!(param.pat.kind, hir::PatKind::Wild))
```

Comment:
slight enhancement: I think you could also mark it as machine-applicable if all the patterns are identifiers and the identifiers are never used in the function body. That would catch things like `.unwrap_or_else(|_err| 1)`.

Reply:
It turns out bindings are already known to be unused since this is a guarded condition above. So I will just include `Binding` in this `matches!`.